### PR TITLE
API Updates & Documentation

### PR DESCRIPTION
**Pull request by michaeldeongreen**
* Changed ValidatePage to OpenGraph
* Updated related Unit Tests
* Re-factiore OpenGraphService
* Updated theurlist_collection.json Validation Page section with new name and request body format
* Updated documentation & images

### CODE REVIEW

File: api/src/LinkyLink/Controllers/LinksController.cs [hunk 1]
```diff
@@ -87,6 +87,24 @@ public async Task<ActionResult<LinkBundle>> GetLinkBundlesForUser(string userId)
         /// <summary>
         /// Create one or more LinkBundles.
         /// </summary>
+        /// <remarks>
```
Comment (Humoiz) [inline api/src/LinkyLink/Controllers/LinksController.cs]: I think its not necessary  to provide an example here for the payload. There is already a payload example present in existing json files and validate-api.http

Comment (michaeldeongreen) [reply api/src/LinkyLink/Controllers/LinksController.cs]: I placed the annotation on there mainly for the Swagger UI.  In the Swagger UI the user will see the request body schema denoted in annotation and can use it to contact the end point.

File: api/src/LinkyLink/Service/OpenGraphService.cs [hunk 2]
```diff
@@ -10,11 +9,29 @@
...
         {
-            string url = singleLinkItem.url, id = singleLinkItem.id;
+            IEnumerable<OpenGraphResult> allResults =
+                await Task.WhenAll((openGraphRequests)
+                .Select(item => GetGraphResult(req, item)));
+
+            return allResults;
+        }
+
+        private async Task<OpenGraphResult> GetGraphResult(HttpRequest req, OpenGraphRequest openGraphRequest)
```
[the remark this answers is not in the document]
Comment (michaeldeongreen) [reply api/src/LinkyLink/Service/OpenGraphService.cs]: I started down the road of re-factor but backed off to simply the PR as it was just for re-naming.  I believe that Service layer classes should not know about HTTP object like HttpRequest and like you stated, the parameter should be a simple type.  I plan open a task to re-factor.

File: api/src/LinkyLink/Controllers/OpenGraphController.cs [hunk 3]
```diff
@@ -32,31 +29,22 @@ public ValidatePageController(IOpenGraphService openGraphService)
...
-                    // expecting a JSON array of objects with url(string), id(string)
-                    IEnumerable<OpenGraphResult> result = await _openGraphService.GetMultipleGraphResults(Request, data);
-                    return new OkObjectResult(result);
-                }
-                else if (data is JObject)
+                if (openGraphRequests != null && openGraphRequests.Count() > 0)
                 {
-                    // expecting a JSON object with url(string), id(string)
-                    OpenGraphResult result = await _openGraphService.GetGraphResult(Request, data);
+                    string requestHost = Request.Host.HasValue ? Request.Host.Host : string.Empty;
```
[the remark this answers is not in the document]
Comment (michaeldeongreen) [reply api/src/LinkyLink/Controllers/OpenGraphController.cs]: This code should not be here, it was erroneously left over for when I went down the re-factor road to enable the OpenGraphService to be testable.  I am not sure if there is a valid use case but the logic was confusing when I saw it.

File: api/src/LinkyLink/Controllers/OpenGraphController.cs [hunk 4]
```diff
@@ -32,31 +29,22 @@ public ValidatePageController(IOpenGraphService openGraphService)
...
-                dynamic data = JsonConvert.DeserializeObject(requestBody);
-
-                if (data is JArray)
-                {
-                    // expecting a JSON array of objects with url(string), id(string)
-                    IEnumerable<OpenGraphResult> result = await _openGraphService.GetMultipleGraphResults(Request, data);
-                    return new OkObjectResult(result);
-                }
-                else if (data is JObject)
+                if (openGraphRequests != null && openGraphRequests.Count() > 0)
```
[the remark this answers is not in the document]
Comment (michaeldeongreen) [reply api/src/LinkyLink/Controllers/OpenGraphController.cs]: Create User Story to refactor the OpenGraphService, which would include the Controller.  This PR was for re-naming the ValidatePageController and making it work with Swagger:
https://dev.azure.com/cseeest/OpenHack/_workitems/edit/1008

File: api/src/LinkyLink/Controllers/OpenGraphController.cs [hunk 5]
```diff
@@ -32,31 +29,21 @@ public ValidatePageController(IOpenGraphService openGraphService)
...
-        ///     }
+        ///        "url": "https://www.microsoft.com"
+        ///     }]
         ///
         /// </remarks>        
-        // POST: api/ValidatePage
+        // POST: api/OpenGraph
         [HttpPost]
-        public async Task<ActionResult<OpenGraphResult>> Post()
+        public async Task<ActionResult<OpenGraphResult>> Post(IEnumerable<OpenGraphRequest> openGraphRequests)
```
Comment (Humoiz) [inline api/src/LinkyLink/Controllers/OpenGraphController.cs]: Previously this end point supported both the array of objects and a object. I dont how the front end works but FE may need to be updated to uptake this change to always send an array in the body. Let Jafar know or create a new task to track that.

Comment (michaeldeongreen) [reply api/src/LinkyLink/Controllers/OpenGraphController.cs]: Created new User Story to track this change.  Has detailed information on what was changed on BE and what change is needed on FE.
https://dev.azure.com/cseeest/OpenHack/_workitems/edit/1009

**Pull request merged**